pkg/controller/zerocapacity/zero_controller.go

```diff
@@ -153,10 +155,15 @@ func (z *Controller) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 
 	phase := instance.Phase()
+
 	switch phase {
 	case "":
 		return reconcile.Result{}, instance.UpdatePhase(ZeroInitializing, nil)
 	case ZeroInitializing:
+		// Perform any transformations required on the CR for backwards-compatibility. Returning if a tranformation or error occurs
```

Comment:
Seems to be we need to call `Transform` as early as possible. on the initial stage before `ZeroInitializing`.

Reply:
Updated :+1: